docs/run-platform-server/configuring.mdx

```diff
@@ -0,0 +1,162 @@
+
+You run multiple instances of `stellar-horizon` each performs a subset of the roles.
+
+### Ingestion Role Instance
+
+You need to allocate one instance to perform ongoing ingestion to capture network activity. This will default to limiting storage of ingested network activity in the database to a sliding window of the last 30 days.
+
+** TODO - make --history-retention-count default to 518400 **
+
+** TODO - make --captive-core-use-db=true by default **
+
+| flag        | envvar         | example                              |
+| ----------- | -------------- | ------------------------------------ |
+| `--db-url`  | `DATABASE_URL` | postgres://localhost/horizon_testnet |
+| `--network` | `NETWORK`      | pubnet                               |
+
+### API Role Instance
+
+You can run none or multiple instances to serve read only API requests.
+
+**TODO - create --tx-sub-enabled flag, default=false. Should return 405 on POSTs to /tx if enabled. this allows skipping need to define --stellar-core-url on api servers since instance doesn't need to be aware of core.**
+
+| flag               | envvar           | example                              |
+| ------------------ | ---------------- | ------------------------------------ |
+| `--db-url`         | `DATABASE_URL`   | postgres://localhost/horizon_testnet |
+| `--tx-sub-enabled` | `TX_SUB_ENABLED` | false                                |
+| `--ingest`         | `INGEST`         | false                                |
+
+### Transaction Submission Role Instance
+
+You can run none or multiple instances to serve transaction submission requests. If you run an instance with transaction submission enabled, your deployment is required to perform the ingestion role on the same database referenced here. Horizon transaction submission depends on ingestion to confirm tx submission status.
+
+If ingestion is planned to be done on a separate instance, add `INGEST=false` on this instance, otherwise don't include the parameter, Horizon will default to `INGEST=true` if absent.
+
+**TODO - Confirm Horizon can use captive core to send tx sub requests and skip stellar-core-url?. **
+
+**TODO - If --tx-sub-enabled=true, horizon should check at startup that ingestion is happening on it's db or exit startup with error.**
+
+| flag               | envvar           | example                              |
+| ------------------ | ---------------- | ------------------------------------ |
+| `--db-url`         | `DATABASE_URL`   | postgres://localhost/horizon_testnet |
+| `--network`        | `NETWORK`        | pubnet                               |
+| `--tx-sub-enabled` | `TX_SUB_ENABLED` | true                                 |
+| `--ingest`         | `INGEST`         | false                                |
+
+## Notes
+
+- In all configuration examples, the same parameter can be provided either as command line argument or an environment variable. If both are provided, the command line takes precedence.
```

Comment:
This clarification is great! I was wondering what happens if I specify both the command line and environment variable when setting up horizon first time locally. Also, I was curious if there are any advantages to choose one approach over the other and may be that's the only one we mention in our documentation (?).

Reply:
we suggest to use env variables across the board in 'passing config' section:

https://github.com/stellar/stellar-docs/pull/162/files#diff-a2be184a631b87a98857de0dd0abe901235c4639179be7c62719b5ab4e6502ddR100

I should add more context there to explain why, i.e. env variables are preferred all the time as they are more secure for obfuscating sensitive info that only the internal process can see, whereas command line leaves a trail that can be observed outside the process. It also depends on the install type, as far as what the operator/user interacts with, for bare metal, they define the env variables, but for containers, they use the container runtime mechanisms to pass env variables through such as docker/helm. 

@jacekn , in the case of Helm chart for Horizon, assuming install params would likely be passed as values file to `helm install ...` correct?

Reply:
@sreuland yeah that right. We tried to make the chart default work out of the box with ability to tune them.
The chart is in early stages though so it can absolutely be improvement and defaults tuned if they are not right. They were based on what we run at SDF but those might not necessarily be the best ones for the community.

Reply:
ok, @urvisavla , can we capture this as dev ticket, 'deprecate command line input of horizon flags' , with an outcome of warnings printed by horizon if any cli present?

I'll do pass through here to rename ref's to any cli format of flags, to the environment variable format.

Reply:
except for `--help` , we should allow just that one cli flag